src/test/scala/com/fasterxml/jackson/module/scala/deser/VavrDeserializerTest.scala

```diff
@@ -0,0 +1,52 @@
+package com.fasterxml.jackson.module.scala.deser
+
+import com.fasterxml.jackson.module.scala.DefaultScalaModule
+import com.fasterxml.jackson.module.scala.deser.EitherJsonTest.{BaseHolder, EitherField, Impl, PlainPojoObject}
+
+/**
+  * Ensuring we can read serialized Vavr Either types into their equivalent Scala types.
+  */
+class VavrDeserializerTest extends DeserializerTest with EitherJsonTestSupport {
+
+  val module: DefaultScalaModule.type = DefaultScalaModule
+
+  "DefaultScalaModule" should "be able to deserialize right with string" in {
+    deserialize[Either[_, String]](s"""["right", "$str"]""")
```

Comment:
why do we need to support either represented as an array when we already support the more logical `{"r":"$str"}`

Reply:
@pjfanning Because that's how the Vavr Jackson module serializes their Either types, and this PR is about adding compatibility for handling that format. [This comment goes into some more detail about the purpose for this change](https://github.com/FasterXML/jackson-module-scala/pull/311#issuecomment-479933407), if you're interested.

Reply:
that comment above does not explain why you want to support the json array format - I would be open to supporting `{"right":"$str"}` as well as `{"r":"$str"}` but I do not support the idea of supporting `["right", "$str"]` (or `["r", "$str"]`)

Reply:
@pjfanning I think it does explain, but it sounds like I worded it badly. Let me try again.

[As you can see here](https://github.com/vavr-io/vavr-jackson/blob/master/src/main/java/io/vavr/jackson/datatype/serialize/EitherSerializer.java), the Vavr Jackson module serializes an Either as an array. That means to resolve issue #303 requires being able to handle deserializing from the array-style format.

The other alternative would be to close this issue and PR as "Won't Do", but I think that would be a bad idea for a number of reasons:

1. The maintainers of the Vavr Jackson module already [merged a similar PR](https://github.com/vavr-io/vavr-jackson/issues/98) that provided deserialization compatibility with the object-style format that this module produces. It was merged under the assumption that the Scala Jackson module would reciprocate, so refusing at this point might come as a surprise.
2. The maintainers of the Vavr Jackson module have expressed interest in using the object-style format instead, but think that handling the array-style format is needed for backwards compatibility. Might be worth pushing them to change, and then eventually reverting support for parsing array-style format a few versions from now.
3. We'd end up in a situation where people can write a Scala Either as JSON and have it read as a Vavr Either, but *could not* write a Vavr Either as JSON and have it read as a Scala Either. That'd be kinda confusing.
4. Last and least, everybody seemed onboard with this (Vavr and Jackson team) when this issue and PR was first opened back in 2017. Unless something changed in the preceding time, it'd have been nicer to object to and close this ticket before we did any work on it. @cowtowncoder, please keep me honest here - if you were okay with this ticket because you thought it was about something else, then that'd be on me instead.

Regardless, it's your call whether or not you'll add reciprocal support for handling Vavr's Either. No matter your decision, thanks for taking the time to review this and maintaining this kickass library.

Reply:
I do not decide things on Scala module, and would defer to PJ, Nate and Morten -- all of who have commented at some point -- but I can see both sides. To me it seems reasonable to support alternative "legacy" format, with strict checking, but not produce it via serialization. Meaning that only very specific kind of array (2 elements, first as matching id) would be accepted, others rejected.
Personally I am more lenient in cases of interoperability; if it wasn't for that part, I would not like the idea of adding another representation. But due to existing usage, wish for backwards compatibility (even tho it is not for older Scala module), I think this would be ok.

@pjfanning does this make more sense?

Reply:
Merged - I confirmed that vavr uses that strange array representation for Either.